### DISCUSSION TITLE
Migrate data to another Outline docker instance (SOLVED)

### DISCUSSION THREAD

**Discussion by rpedrica** [outline/outline#4025] [General] [answered]
I've been using a local Outline instance for some time but would like to put it at public location. I've copied my outline docker folder to a VPS and adjusted all configs to the new url. The main Outline site comes up with the login screen and I select the "Continue with OpenID" option (as per my local instance) but get an error:

```
OIDC Server

Redirect URI Error
The request fails due to a missing, invalid, or mismatching redirection URI (redirect_uri).
```

I've checked all configs and can't see where I'm going wrong.

Nonetheless, I thought of re-initializing this instance and then migrating data so I can get my docs back. Can anyone advise how I would go about doing this? Maybe:

1. just copy over the pgsql folders?
2. do a pgsql export and import?

This would be useful for backups too.

A note as well that my docker script is not from the official outline documentation (https://wiki.generaloutline.com/share/770a97da-13e5-401e-9f8a-37949c19f97e/doc/docker-7pfeLP5a8t) - in fact I can't find where I got it originally.

Any assistance would be appreciated.

**Comment by tommoor** [comment]
This error would suggest misconfiguration on the OIDC provider, you probably need to add the new public url as a valid post-auth redirect url

**Comment by rpedrica** [comment]
Thanks @tommoor

I've got further as follows - if I configure Outline with a direct public URL (ie. not through NPM) then I get it (sort of) working using that URL:

http://host.domain.com:8888/

- I get presented with the "Continue with OpenID" message
- click and get the "Redirect URI Error" but there is a Login option in the top right corner which I click
- Login with creds (username and password as previously configured in my local instance) - I am successfully logged in and can access my wiki data. So not perfect but at least I can get in.

Now if I change the URLs in env.outline and env.oidc to use my NPM/load balancer URL (https://outline.domain.com/) and at the point that I get to the "Redirect URI Error" page, I click on the Login in top right corner and enter creds, however now it just takes me back to the original "Continue with OpenID" Message.

If I do this again, then the top right corner I get Admin and Logout so the creds worked to some degree but OIDC is getting confused somewhere or I don't know where to correctly config the URLs for public access. The Admin link takes me to the oidc server user/groups/etc. config screen.

Thanks for your replies, Robby

**Comment by rpedrica** [comment, marked as answer]
And 1 min after posting this I found the solution - in the oidc-server admin section, I was checking the client for outline and noted that the url for the non-NPM access had been placed there by the initiating script. Changed it to the NPM url and all working fine now. Marking as resolved.
